pkg/encoding/v1/encoding_test.go

```diff
@@ -397,6 +465,54 @@ volumes: []
 			"",
 			nil,
 		},
+		{
+			true, `
+version: 0.1-dev
+services:
+- name: helloworld
+  replicas: 2
+  containers:
+  - image: tomaskral/nonroot-nginx
+`,
+			&object.OpenCompose{
+				Version: Version,
+				Services: []object.Service{
+					{
+						Name:     "helloworld",
+						Replicas: &positiveReplicas,
+						Containers: []object.Container{
+							{
+								Image: "tomaskral/nonroot-nginx",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			true, `
+version: 0.1-dev
+services:
+- name: helloworld
+  replicas: -1
```

Comment:
even after giving value as negative this test passes, since the check on negativity happens later? Apart from adding tests about createdeployment and verifying this actually fails @tnozicka do you have any thought on handling this well?